src/tiledb/cloud/vector_search/file_ingestion.py

```diff
@@ -0,0 +1,354 @@
+from typing import Dict, List, Mapping, Optional, Sequence
+
+from tiledb.cloud import dag
+from tiledb.cloud.utilities import as_batch
+
+
+def ingest_files_dag(
+    file_dir_uri: str,
+    index_uri: str,
+    file_name: Optional[str] = None,
+    acn: Optional[str] = None,
+    config=None,
+    environment_variables: Mapping[str, str] = {},
```

Comment:
Including mutable objects as default parameters will have unexpected results, since the *reference* is used as the default parameter:

```python
def do_something(it: dict = {}):
  return it, len(it)

a_dict, size = do_something()
a_dict["my value"] = 1

dict_two, size_two = do_something()
# dict_two == {"my value": 1}; size_two == 1
# a_dict is dict_two
```

Instead, I would recommend using an Optional:

```python
def do_something(it: Optional[dict] = None):
  it = {} if it is none else it
  return it, len(it)

a_dict, size = do_something()
a_dict["my_value"] = 1

dict_two, size_two = do_something()
# dict_two == {}; size_two == 0
# a_dict is not dict_two
```

Reply:
Applied